clang/lib/Sema/SemaOverload.cpp

```diff
@@ -5450,6 +5450,12 @@ TryReferenceInit(Sema &S, Expr *Init, QualType DeclType,
       return ICS;
     }
 
+    if (isRValRef && ICS.UserDefined.ConversionFunction->getReturnType()
```

Comment:
It isn't clear to me why/if this is the right solution here.  Do we have justification here based on the standard that we can put the standards quote here?  Also, this is a bit of an awkward place to put this, so it doesn't seem clear to me that this is even the right place.

Unfortunately i'm not particularly knowledgable about this overload code, so hopefully someone more knowledgable comes along.